### PR TITLE
IGNITE-23708 testAutomaticRebalanceIfMajorityIsLost is flaky

### PR DESCRIPTION
**Pull request by JAkutenshi**
JIRA Ticket: [IGNITE-23708](https://issues.apache.org/jira/browse/IGNITE-23708)

## The goal

The main goal of the PR is to fix `testAutomaticRebalanceIfMajorityIsLost`:

1. Reset rebalance may be slower a little than expected 2s timeout on `assertRealAssignments(node0, partId, 1)` especially after [IGNITE-22036](https://issues.apache.org/jira/browse/IGNITE-22036) changes.
2. There is unnecessary planned rebalance with the same planned assignments as pending force reset assignments set equals to `[1]`. This leads also to a flacky non-null result on `assertNull(getPendingAssignments(node0, partId))` check.

## The reason

Tests shouldn't be flacky and flawed implementation should be fixed.

## The solution

1. `assertRealAssignments`'s timeout is increased up to 5s.
2. If during `GroupUpdateRequest#partitionUpdate` if there are alive nodes we have equal `part` and `stable` assignments we shouldn't schedule the redundant rebalance and then should put just `null` then.

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java [hunk 1]
```diff
@@ -304,14 +304,19 @@ private static CompletableFuture<Integer> partitionUpdate(
 
         Assignment nextAssignment = nextAssignment(localPartitionStateMessageByNode, partAssignments);
 
+        boolean isNextEqualsToRestoredAsPlannedAssignments = partAssignments.size() == 1 && partAssignments.contains(nextAssignment);
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java:307]: I believe that something like `isProposedPendingEqualsProposedPlanned` would be clearer. Up to you.

Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java:307]: How it's possible to have nextAssignment not in partAssignment? It it's not, I'd rather add an assert.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java [hunk 2]
```diff
@@ -304,14 +304,19 @@ private static CompletableFuture<Integer> partitionUpdate(
...
+
         // There are nodes with data, and we set pending assignments to this set of nodes. It'll be the source of peers for
         // "resetPeers", and after that new assignments with restored replica factor wil be picked up from planned assignments
         // for the case of the manual update, that was triggered by a user.
         Iif invokeClosure = prepareMsInvokeClosure(
                 partId,
                 longToBytesKeepingOrder(revision),
                 Assignments.forced(Set.of(nextAssignment), assignmentsTimestamp).toBytes(),
-                Assignments.toBytes(partAssignments, assignmentsTimestamp)
+                // In case if nodes set is equal then we shouldn't schedule the same planned rebalance
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java:316]: is equal to what?

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java (hunk 1 above)
Comment (JAkutenshi) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java]: Agree

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java (hunk 2 above)
Comment (JAkutenshi) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java]: Old commentary, I guess the new one is more precise now

**Review by sanpwc — APPROVED** [review]

**Pull request merged**